interface/src/app/analyze/analysis/sampleBin.js

```diff
@@ -77,6 +80,26 @@ function($log, $cacheFactory, $q, Sample, Activity) {
       }
     },
 
+    getSampleGroups: function() {
+      var keys = Object.keys(this.sampleGroups);
+      var samplesByGroup = {};
+      var i;
+
+      // each distinct value in sampleGroups becomes a key in samplesByGroup,
+      // and the keys of sampleGroups are collected in a list within each
+      // corresponding value of samplesByGroup
+      for (i = 0; i < keys.length; i++) {
+        if (samplesByGroup[this.sampleGroups[+keys[i]]]) {
```

Comment:
I think this code is definitely good as it is, but I am just curious of what you all think of using a variable inside this for-loop to hold the value of what ``+keys[i]`` is, just to make it more readable. Maybe something like ``var groupID = +keys[i]`` or ``var groupNumber = +keys[i]``? What do you think?